lib/src/repo.rs

```diff
@@ -318,15 +350,30 @@ pub struct RepoLoader {
 }
 
 impl RepoLoader {
-    pub fn init(user_settings: &UserSettings, repo_path: &Path) -> Self {
+    pub fn init(
+        user_settings: &UserSettings,
+        repo_path: &Path,
+        backend_factories: &BackendFactories,
+    ) -> Self {
         let store_path = repo_path.join("store");
-        let git_target_path = store_path.join("git_target");
-        let backend: Box<dyn Backend> = if git_target_path.is_file() {
-            Box::new(GitBackend::load(&store_path))
-        } else {
-            Box::new(LocalBackend::load(&store_path))
+        let backend_type = match fs::read_to_string(store_path.join("backend")) {
+            Ok(content) => content,
+            Err(_) => {
+                // For compatibility with existing repos. TODO: Delete in spring of 2023 or so.
```

Comment:
nit: safer to check the error kind, but I don't care much since fs::write() would also fail if the error were e.g. EACCES.